core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -296,6 +306,12 @@ public String createBlock(long sessionId, long blockId, int tier,
           sessionId, blockId, tier,
           createBlockOptions.getMedium(), createBlockOptions.getInitialBytes(), e);
 
+      // mAddress is null if the worker is not started
+      if (mAddress == null) {
+        throw new WorkerOutOfSpaceException(ExceptionMessage.CANNOT_REQUEST_SPACE
+                .getMessage(mWorkerId.get(), blockId), e);
+      }
+
```

Comment:
This is to ensure that when `DefaultBlockWorker` is not started, as in unit-test, this method still throws reasonable Exception rather than a `NullPointerException` caused by `mAddress` being `null`

Reply:
How about using Preconditions.checkState(mAddress != null, "Worker has not been started" )?  Using the  WorkerOutOfSpaceException and the message of `CANNOT_REQUEST_SPACE` does not make sense to me.

Reply:
We intend to check that `createBlock` throws `WorkerOutOfSpaceException` in case when there's no enough space. But in testing environment, `mAddress` is `null` because worker is not started. If we use `checkState` here, we will get an 'IllegalStateException' in the test method, which I think is not ideal..